x/authenticator/ante/authenticator.go

```diff
@@ -57,90 +59,117 @@ func (ad AuthenticatorDecorator) AnteHandle(
 
 	feeTx, ok := tx.(sdk.FeeTx)
 	if !ok {
-		return ctx, errorsmod.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+		return ctx, errorsmod.Wrap(sdkerrors.ErrTxDecode, "transaction must be a FeeTx")
 	}
 
-	// The fee payer by default is the first signer of the transaction
+	// Determine the fee payer, which defaults to the first signer of the transaction.
 	feePayer := feeTx.FeePayer()
-	feePayerAuthenticated := false
-
-	// Recover from any OutOfGas panic to return an error with information of the gas limit having been reduced
-	defer func() {
-		if r := recover(); r != nil {
-			if feePayerAuthenticated {
-				panic(r)
-			}
-			switch r.(type) {
-			case sdk.ErrorOutOfGas:
-				log := fmt.Sprintf(
-					"FeePayer not authenticated yet. The gas limit has been reduced to %d. Consumed: %d",
-					ad.maxFeePayerGas, payerGasMeter.GasConsumed())
-				err = sdkerrors.Wrap(sdkerrors.ErrOutOfGas, log)
-			default:
-				panic(r)
-			}
-		}
-	}()
```

Comment:
When we fail due to running out of gas for the fee payer... are we getting the proper error? This was needed before because the default use the original gas limit in the error message (it's caching it)

Reply:
Good catch, I ommited this on purpose but adding it back in back as it improves UX a lot.